CHANGELOG.md

```diff
@@ -4,6 +4,25 @@
 
 ### Changes
 
+#### Token Introspection
+
+The introspection endpoint has been fixed in case of the encoding like mentioned in bugfixes.  
+Additionally, authorization has been added to this endpoint. It will now make sure that the request also includes
+an `AUTHORIZATION` header with either a valid `Bearer JwtToken` or `Basic B64EncodedClientId:ClientSecret` to prevent
+token scanning.
+
+The way of authorization on this endpoint is not really standardized, so you may run into issues with your client
+application. If so, you can disable the authentication on this endpoint with
+
+```
+# Can be set to `true` to disable authorization on `/oidc/introspect`.
+# This should usually never be done, but since the auth on that endpoint is not
+# really standardized, you may run into issues with your client app.
+# If so, please open an issue about it.
+# default: false
+DISABLE_INTROSPECT_AUTH=true
+```
```

Comment:
For anyone in future landing here, the only implementation I know of that doesn't have basic auth by default is [Ory Hydra](https://www.ory.sh/docs/hydra/self-hosted/production#exposing-administrative-and-public-api-endpoints).

The referenced lack of standardized endpoint protection is covered in [RFC 7662 - Section 2.1](https://datatracker.ietf.org/doc/html/rfc7662#section-2.1), where protecting the endpoint is required but how that is approached is flexible. A benefit of opt-out support is that you can implement alternative support such as mTLS.

I'm not sure about clients, as basic auth would always be valid AFAIK as part of the URL (_although [some clients are deprecating support](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#access_using_credentials_in_the_url)_). For `/introspect` this would be the only way for [Dovecot](https://doc.dovecot.org/configuration_manual/authentication/oauth2/#full-config-file) to pass credentials for authorization (_the [`POST` request doesn't use the `client_id` / `client_secret` settings](https://github.com/dovecot/core/blob/03316d828c31ec736d87ba51e494bcef7a259b56/src/lib-oauth2/oauth2-request.c#L323-L330), only the [`introspection_mode = auth` setting supports adding the `Authorization: Bearer` header](https://github.com/dovecot/core/blob/03316d828c31ec736d87ba51e494bcef7a259b56/src/lib-oauth2/oauth2-request.c#L250-L267) which is a `GET` request for `/userinfo` not `/introspect`_).

There may be other clients 😅